Add layer-cycle example (#19069)
qmk/qmk_firmware: docs/feature_layers.md

}
```

### Example: Keycode to cycle through layers

This example shows how to implement a custom keycode to cycle through a range on layers.

```c
// Define the keycode, `SAFE_RANGE` avoids collisions with existing keycodes
enum keycodes {

>   KC_CYCLE_LAYERS = SAFE_RANGE,

Reviewer: ```suggestion
  KC_CYCLE_LAYERS = QK_USER,
```

};

// 1st layer on the cycle
#define LAYER_CYCLE_START 0
// Last layer on the cycle
#define LAYER_CYCLE_END   4

// Add the behaviour of this new keycode
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
  switch (keycode) {
    case KC_CYCLE_LAYERS:
      // Our logic will happen on presses, nothing is done on releases
      if (!record->event.pressed) { 
        // We've already handled the keycode (doing nothing), let QMK know so no further code is run unnecessarily
        return false;
      }

      uint8_t current_layer = get_highest_layer(layer_state);

      // Check if we are within the range, if not quit

>       if (curent_layer > LAYER_CYCLE_END || current_layer < LAYER_CYCLE_START) {

Reviewer 2: `      if (current_layer > LAYER_CYCLE_END || current_layer < LAYER_CYCLE_START) {`
small typo here

}
```

### Example: Keycode to cycle through layers

> This example shows how to implement a custom keycode to cycle through a range on layers.

Reviewer 3: ```suggestion
This example shows how to implement a custom keycode to cycle through a range of layers.
```

```c
// Define the keycode, `SAFE_RANGE` avoids collisions with existing keycodes
enum keycodes {
  KC_CYCLE_LAYERS = QK_USER,
};

// 1st layer on the cycle
#define LAYER_CYCLE_START 0
// Last layer on the cycle
#define LAYER_CYCLE_END   4

// Add the behaviour of this new keycode
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
  switch (keycode) {
    case KC_CYCLE_LAYERS:
      //our logic will happen on presses, nothing is done on releases
      if (!record->event.pressed) { 

>         //already handled the keycode, let QMK know it so no further code is run uselessly

Reviewer 3: ```suggestion
        // already handled the keycode, let QMK know so no further code is run unnecessarily.
```

        return false;
      }

      uint8_t current_layer = get_highest_layer(layer_state);

      // Check if we are within the range, if not quit
      if (curent_layer > LAYER_CYCLE_END || current_layer < LAYER_CYCLE_START) {
        return false;
      }

>       // If we are in range, turn next layer on without affecting other layers, so we don't break `KC_TRNS`

Reviewer 3: Also perhaps elaborate what you mean by this?
```suggestion
      // If we are in range, activate next layer without affecting other layers, so we don't break `KC_TRNS`
```

      if (current_layer < LAYER_CYCLE_END) {

>           layer_on(current_layer+1);

Reviewer 3: ```suggestion
          layer_on(current_layer + 1);
```

          return false;
      }

      // Getting here means we are on the last layer of the cycle

>       layer_move(LAYER_CYCLE_START); //move to starting layer, ie turn OFF all other layers (default one is always ON, tho)

Reviewer 3: ```suggestion
      layer_move(LAYER_CYCLE_START); // move to starting layer, i.e. turn OFF all other layers (default one is always ON, though)
```

      return false;

>     default: //don't affect other keycodes
>       return true; //returning true means QMK will handle the keycode

Reviewer 3: ```suggestion
    default:
      return true; // process other keycodes normally
```

  }
}

> // Place this new keycode wherever you want

Reviewer 3: ```suggestion
// Place `KC_CYCLE_LAYERS` as a keycode in your keymap
```

}
```

### Example: Keycode to cycle through layers

This example shows how to implement a custom keycode to cycle through a range of layers.

```c
// Define the keycode, `QK_USER` avoids collisions with existing keycodes
enum keycodes {
  KC_CYCLE_LAYERS = QK_USER,
};

// 1st layer on the cycle
#define LAYER_CYCLE_START 0
// Last layer on the cycle
#define LAYER_CYCLE_END   4

// Add the behaviour of this new keycode
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
  switch (keycode) {
    case KC_CYCLE_LAYERS:
      // Our logic will happen on presses, nothing is done on releases
      if (!record->event.pressed) { 
        // We've already handled the keycode (doing nothing), let QMK know so no further code is run unnecessarily
        return false;
      }

      uint8_t current_layer = get_highest_layer(layer_state);

      // Check if we are within the range, if not quit
      if (curent_layer > LAYER_CYCLE_END || current_layer < LAYER_CYCLE_START) {
        return false;
      }

      // If we are within the range, but not on the last layer, turn next layer ON
      if (current_layer < LAYER_CYCLE_END) {

>           layer_on(current_layer + 1);

Reviewer: layer_move might make more sense here.  Cycling, to me, implies that only one of the layers is on, and that we're cycling through which layer is on.  So, if there are transparencies in these layers, it may cause some confusion

Also, this way, you could use this for this block and the next: 
```c
    current_layer++;
    if (current_layer > LAYER_CYCLE_END) {
        current_layer = LAYER_CYCLE_START;
    }
    layer_move(current_layer);
```
